lib/charms/layer/caas_base.py

```diff
@@ -15,6 +15,11 @@ def pod_spec_set(spec):
         run(['pod-spec-set'], stdout=PIPE, stderr=PIPE, check=True, input=spec.encode('utf-8'))
     except CalledProcessError as err:
         log(f'pod-spec-set encountered an error: {err.stderr.decode("utf-8")}', level='ERROR')
+
+        if err.stderr == b'ERROR application jupyterhub not alive\n':
```

Comment:
This is application specific.  Should check this against a regular expression instead.